DSpace/dspace-angular
# Issue #1381 - DSO sub route titles fix

## References
Fixes #1381 

## Description
This PR fixes page titles (browser tab) to display the correct title in order of priority (from highest to lowest): Translated i18n "title" key from route data > Name of the DSO found in route data

## Instructions for Reviewers
Changes:
Modified MetadataService's `processRouteChange()` to update meta tags with the dso from the route first, then overwrite the title meta tag with the translated title from the route's data (this happened in reverse before this change)

How to test:
* Visit any page not part of a DspaceObject route (e.g. /home, /search...)
* Confirm the configured route's title is translated and displayed in the browser's tab
* Visit any page of a DspaceObject route (e.g. community, collection, item page)
* Confirm the object's name is displayed in the browser's tab
* Visit sub-routes of any of these objects (e.g. /edit, collection/itemtemplate...)
* Confirm the configured route's title is translated and displayed in the browser's tab (if one is configured, otherwise it should show the object's name!)

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_
